### ISSUE TITLE
Django-storages S3 boto backend throws AttributeError on is_outdated()

### ISSUE THREAD

**Issue by yuvadm** [cyberdelia/django-pipeline#329]
When using the following storage backend, from django-storages:

```python
from storages.backends.s3boto import S3BotoStorage

class S3PipelineCachedStorage(PipelineMixin, CachedFilesMixin, S3BotoStorage):
    pass
```

this error is thrown during `collectstatic`:

```
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 118, in collect
    for original_path, processed_path, processed in processor:
  File "/app/.heroku/python/lib/python2.7/site-packages/pipeline/storage.py", line 32, in post_process
    packager.pack_stylesheets(package)
  File "/app/.heroku/python/lib/python2.7/site-packages/pipeline/packager.py", line 94, in pack_stylesheets
    variant=package.variant, **kwargs)
  File "/app/.heroku/python/lib/python2.7/site-packages/pipeline/packager.py", line 103, in pack
    paths = self.compile(package.paths, force=True)
  File "/app/.heroku/python/lib/python2.7/site-packages/pipeline/packager.py", line 97, in compile
    return self.compiler.compile(paths, force=force)
  File "/app/.heroku/python/lib/python2.7/site-packages/pipeline/compilers/__init__.py", line 49, in compile
    return list(executor.map(_compile, paths))
  File "/app/.heroku/python/lib/python2.7/site-packages/concurrent/futures/_base.py", line 549, in map
    yield future.result()
  File "/app/.heroku/python/lib/python2.7/site-packages/concurrent/futures/_base.py", line 404, in result
    return self.__get_result()
  File "/app/.heroku/python/lib/python2.7/site-packages/concurrent/futures/_base.py", line 356, in __get_result
    raise self._exception
AttributeError: 'NoneType' object has no attribute 'last_modified'
```

**Comment by viciu** [comment]
It looks like Django and Pipeline incompatibility.
Issue is gone when using Django < 1.7 and Pipeline 1.3.27.

**Comment by skirsdeda** [comment]
This happens simply because outfile in is_outdated does not exist yet. A simple fix is to check if outfile path exists in storage and return True if that's the case.

**Comment by cyberdelia** [comment]
Fixed by #472
